src/main/java/org/jmxtrans/agent/Query.java

```diff
@@ -220,11 +220,12 @@ private void collectAndExportAttribute(MBeanServer mbeanServer, OutputWriter out
                     value = compositeData.get(key);
                 }
             } else if (attributeValue instanceof Map) {
-                Map<String, Object> mapData = (Map<String, Object>) attributeValue;
+                //The attribute key could be other types, e.g. Double/int
+                Map<Object, Object> mapData = (Map<Object, Object>) attributeValue;
                 if (key == null) {
                     // Get for all keys
-                    for (Map.Entry<String, Object> entry : mapData.entrySet()) {
-                        String key = entry.getKey();
+                    for (Map.Entry<Object, Object> entry : mapData.entrySet()) {
```

Comment:
Instead of a specific one-off to support doubles, could we instead just `toString` the key before adding/getting and keep the key as a `String`? That seems cleaner and would work for other types.

Reply:
I believe this already supports all types since Java Objects must implement `toString()`. (`Double` is just an example type covered in tests)

Reply:
ic. we're just casting the mbean map type more generically. I initially thought this change was being exposed more so than it is. My bad.